corehq/apps/app_manager/suite_xml.py

```diff
@@ -1943,6 +1944,46 @@ def get_target_module(case_type, module_id, with_product_details=False):
             except IndexError:
                 pass
 
+        self.add_parent_datums(datums, module)
+
+        return datums, assertions
+
+    def add_parent_datums(self, datums, module):
+
+        def update_refs(action_, datum_, changed_ids):
+            if action_:
+                # Only advanced module actions have a parent_tag attribute. Basic modules don't need one
+                # because they only deal with one case type. Set parent_tag to None for them.
+                parent_tag = getattr(action_, 'parent_tag', None)
+                if parent_tag in changed_ids:
+                    # update any reference to previously changed datums
+                    change = changed_ids[parent_tag]
+                    nodeset = datum_.nodeset
+                    old = session_var(change['old_id'])
+                    new = session_var(change['new_id'])
+                    datum_.nodeset = nodeset.replace(old, new)
+
+        def avoid_duplicate_ids(action_, this_datum_, parent_datum_, datum_ids_, changed_ids):
+            if action_:
+                case_tag = getattr(action_, 'case_tag', None)  # Set case_tag to None for basic modules
+                if parent_datum_.id in datum_ids_:
+                    # We can't set a new ID to an ID we already have. We need to rename current ID
+                    datum = datum_ids_[parent_datum_.id]
+                    new_id = '_'.join((parent_datum_.id, datum['case_type']))
```

Comment:
Do you mean `new_id = '_'.join((this_datum_.id, datum['case_type']))`? 

When I do that `new_id` is set to `parent_id_guppy` (for example) instead of `case_id_guppy`, because the datum in the parent entry has the id `case_id`, but the datum in this entry has the id `parent_id`.